### PR TITLE
Version changes

### PR DESCRIPTION
**Pull request by piponazo**
See the commit messages for more details. This PR solves #445

### CODE REVIEW

**Review by clanmills — COMMENTED** [review]
I'm confused.  I can't find the branch 'versionChanges' in either exiv2/exiv2 or piponazo/exiv2, so I can't build this.

1) Revision number.

I thought we agreed it would be MAJOR.MINOR.PATCH.TWEAK  and TWEAK=0 is a public build (suitable for external use).  Is it possible to define those as variables in CMakeLists.txt and put them into the project(exiv2 ...).  Then propagate everything into version.cpp

2) Let's get rid of sBuild in Params::version()
I added that about 2010 and then thought of the much more general ` $ version --verbose --version`, so sBuild and versionNumberHexString() are relics.

3) Please remove versionNumberHexString() from the API
```
626 rmills@rmillsmbp:~/gnu/github/piponazo/exiv2 $ finder version.hpp $G Hex $___
    EXIV2API std::string versionNumberHexString();
627 rmills@rmillsmbp:~/gnu/github/piponazo/exiv2 $ 
```
We should do this because we don't need it.  We should do this to ensure that existing applications will be unable to link v0.27 release candidates!

**Comment by piponazo** [comment]
You should be able to find this branch in my fork: `piponazo:versionChanges`. Otherwise, I could not have created the Pull-Request.

> 1. Revision number. I thought we agreed it would be MAJOR.MINOR.PATCH.TWEAK and TWEAK=0 is a public build (suitable for external use). Is it possible to define those as variables in CMakeLists.txt and put them into the project(exiv2 ...). Then propagate everything into version.cpp

That's what I did. In CMakeLists.txt we have `VERSION 0.27.0.0` for the master branch. The moment we decide to have a RC1, we can change this number in the branch we will use for the RC1. 

We can have different approaches for the version numbers. Something that I did in other projects is to have a text file with the version number, and then read that file from CMake for setting up the PROJECT_VERSION. However, all these mechanisms over complicate things, and I think we can live with the current approach for the moment. 

> 2. Let's get rid of sBuild in Params::version(). I added that about 2010 and then thought of the much more general `$ exiv2 --verbose --version`, so sBuild and versionNumberHexString() are relics.

It is not clear for me what you want me to do here. `Params::version(bool verbose)` is implementing both functionalities right now:
```
exiv2 --version
exiv2 --version --verbose
```
Are you proposing to always print the verbose information with `exiv2 --version` (call to `Exiv2::dumpLibraryInfo()` and remove the line 221 from exiv2.cpp ?

> 3. Please remove versionNumberHexString() from the API

I considered to do so, but I noticed that the function is also used in `epsimage.cpp` in the lines 893 and 917. I am not sure what this code is doing and I decided to not take any actions in that topic for the moment. What do you think ? 

> 4 Can we add any unique identifier (such as the date/time or UUID) for builds with TWEAK !=0? This enables us to know the difference between builds. In SVN days, I used the SVN# for this purpose. So every bundle has a different name, even if the version strings are the same.

I guess we can come up with a solution for this problem. But I would prefer to do it in a separate branch, and first we would need to create an issue in Github with a full description of the problem to be solved.

File: include/exiv2/config.h [hunk 1]
```diff
@@ -1,43 +1,12 @@
...
-
 ///// Start of Visual Studio Support /////
 #ifdef  _MSC_VER
 
-#define _MSC_VER_2015 1900
-#define _MSC_VER_2013 1800
-#define _MSC_VER_2012 1700
-#define _MSC_VER_2010 1600
-#define _MSC_VER_2008 1500
-#define _MSC_VER_2005 1400
```
Comment (piponazo) [inline include/exiv2/config.h]: I removed the ones for 2005 and 2003 accidentally. I have to place back those lines.

**Review by clanmills — APPROVED** [review]
@piponazo I'm going to approve this to ensure you're able to integrate this on Friday.  If I see something that needs to be changed, I'll raise an issue.

**Pull request merged**